File: mobile_app/saviya/lib/model/review_model.dart
// model/review_model.dart
class Review {
  final int id;
  final int rating;
  final String comment;
  final int productId;
  final int userId;
  final String createdAt;
  final String userFirstName;
  final String userLastName;

  Review({
    required this.id,
    required this.rating,
    required this.comment,
    required this.productId,
    required this.userId,
    required this.createdAt,
    required this.userFirstName,
    required this.userLastName,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['id'],
      rating: json['rating'],
      comment: json['comment'] ,
      productId: json['productId'], 
      userId: json['userId'],
      createdAt: json['createdAt'],
      userFirstName: json['userFirstName'] ,
      userLastName: json['userLastName'] ,
    );
  }
}

File: backend/models/reviewModel.js
const { DataTypes } = require("sequelize");
const sequelize = require("../connection");
const User = require("./userModel");
const Product = require("./productModel");

const Review = sequelize.define(
    "Review",
    {
        rating: {
            type: DataTypes.INTEGER,
            allowNull: false,
            validate: { min: 1, max: 5 },
        },
        comment: {
            type: DataTypes.TEXT,
            allowNull: true,
        },
        productId: {
            type: DataTypes.INTEGER,
            allowNull: false,
            references: { model: Product, key: "id" },
            onDelete: "CASCADE",
            field: "product_id",
        },
        userId: {
            type: DataTypes.INTEGER,
            allowNull: false,
            references: { model: User, key: "id" },
            onDelete: "CASCADE",
            field: "user_id",
        },

        createdAt: {
            type: DataTypes.DATE,
            defaultValue: DataTypes.NOW,
            field: "created_at"
        },
        updatedAt: {
            type: DataTypes.DATE,
            defaultValue: DataTypes.NOW,
            field: "updated_at"
        }
    },
    {
        tableName: "reviews",
        timestamps: true,
    }
);

Review.belongsTo(User, { foreignKey: "userId", as: "user" });
User.hasMany(Review, { foreignKey: "userId" });

Review.belongsTo(Product, { foreignKey: "productId" });
Product.hasMany(Review, { foreignKey: "productId" });

module.exports = Review;
